aws-apigateway-go-routes/main.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	apigateway "github.com/pulumi/pulumi-aws-apigateway/sdk/go/apigateway"
+	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/cognito"
+	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/iam"
+	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/lambda"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		role, err := iam.NewRole(ctx, "auth-exec-role", &iam.RoleArgs{
+			AssumeRolePolicy: pulumi.String(`{
+				"Version": "2012-10-17",
+				"Statement": [{
+					"Sid": "",
+					"Effect": "Allow",
+					"Principal": {
+						"Service": "lambda.amazonaws.com"
+					},
+					"Action": "sts:AssumeRole"
+				}]
+			}`),
+		})
+		if err != nil {
+			return err
+		}
+
+		logPolicy, err := iam.NewRolePolicy(ctx, "lambda-log-policy", &iam.RolePolicyArgs{
+			Role: role.Name,
+			Policy: pulumi.String(`{
+                "Version": "2012-10-17",
+                "Statement": [{
+                    "Effect": "Allow",
+                    "Action": [
+                        "logs:CreateLogGroup",
+                        "logs:CreateLogStream",
+                        "logs:PutLogEvents"
+                    ],
+                    "Resource": "arn:aws:logs:*:*:*"
+                }]
+            }`),
+		})
+		if err != nil {
+			return err
+		}
+
+		authLambda, err := lambda.NewFunction(ctx, "auth-lambda", &lambda.FunctionArgs{
+			Runtime: lambda.RuntimeGo1dx,
+			Code: pulumi.NewAssetArchive(map[string]interface{}{
+				".": pulumi.NewFileArchive("./bin/authorizer.zip"),
+			}),
+			Handler: pulumi.String("authorizer"),
+			Role:    role.Arn,
+		}, pulumi.DependsOn([]pulumi.Resource{logPolicy}))
+		if err != nil {
+			return err
+		}
+
+		helloHandler, err := lambda.NewFunction(ctx, "hello-handler", &lambda.FunctionArgs{
+			Runtime: lambda.RuntimeGo1dx,
+			Code: pulumi.NewAssetArchive(map[string]interface{}{
+				".": pulumi.NewFileArchive("./bin/handler.zip"),
+			}),
+			Handler: pulumi.String("handler"),
+			Role:    role.Arn,
+		}, pulumi.DependsOn([]pulumi.Resource{logPolicy}))
+		if err != nil {
+			return err
+		}
```

Comment:
Would it make sense to move lambdas to a separate file/function similar to Python?